MapboxDirections/MBVisualInstruction.swift

```diff
@@ -56,15 +62,57 @@ open class VisualInstruction: NSObject, NSSecureCoding {
     public convenience init(json: [String: Any]) {
         let text = json["text"] as? String
-        let maneuverType = ManeuverType(description: json["type"] as! String) ?? .none
-        let maneuverDirection = ManeuverDirection(description: json["modifier"] as! String)  ?? .none
-        let textComponents = (json["components"] as! [JSONDictionary]).map {
-            VisualInstructionComponent(json: $0)
+        let componentsDictionary = json["components"] as? [JSONDictionary]
+        var components = [Component]()
+        
+        var maneuverType: ManeuverType = .none
+        if let type = json["type"] as? String, let derivedType = ManeuverType(description: type) {
+            maneuverType = derivedType
+        }
+        
+        var maneuverDirection: ManeuverDirection = .none
+        if let modifier = json["modifier"] as? String,
+            let derivedDirection = ManeuverDirection(description: modifier) {
+            maneuverDirection = derivedDirection
+        }
+        
+        var componentType: VisualInstructionComponentType = .text
+        if let components = componentsDictionary, let firstComponent = components.first, let type = firstComponent["type"] as? String,
+           let currentComponentType = VisualInstructionComponentType(description: type) {
+            componentType = currentComponentType
+        }
+        
+        if let dictionary = componentsDictionary {
+            if componentType == .lane {
+                components = dictionary.map { record in
+                    let type = VisualInstructionComponentType(description: record["type"] as? String ?? "") ?? .text
+                    var indications = LaneIndication()
+                    if let directions = record["directions"] as? [String],
+                        let laneIndications = LaneIndication(descriptions: directions) {
+                        indications = laneIndications
+                    }
+                    var isUsable = false
+                    if let active = record["active"] as? Bool {
+                        isUsable = active
+                    }
+                    return LaneIndicationComponent(text: record["text"] as? String, type: type, indications: indications, isUsable: isUsable)
+                }
+            } else {
+                components = dictionary.map { record in
+                    let type = VisualInstructionComponentType(description: record["type"] as? String ?? "") ?? .text
+                    let imageURL = URL(string: (record["imageBaseURL"] as? String) ?? "")
+                    return VisualInstructionComponent(type: type,
+                                                      text: record["text"] as? String,
+                                                  imageURL: imageURL,
+                                              abbreviation: record["abbr"] as? String,
+                                      abbreviationPriority: record["abbr_priority"] as? Int ?? NSNotFound)
```

Comment:
Only align parameters by the colon in Objective-C, not in Swift. Use Xcode’s reindent command to reformat this code.